### PR TITLE
Enable encoder settings VIA.

### PR DESCRIPTION
**Pull request by 7-rate**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/pistachio_mp/keymaps/via/keymap.c
Comment (drashna) [inline keyboards/pistachio_mp/keymaps/via/keymap.c:72]: This will miss most keycodes. 

You should use action_exec here instead, as that will work with all keycodes. 

See `keyboards/work_louder/encoder_actions.c` for an example of what I mean.

File: keyboards/pistachio_mp/pistachio_mp.h
Comment (drashna) [inline keyboards/pistachio_mp/pistachio_mp.h:61]: This can be moved to the via keymap file
```suggestion
```

File: keyboards/pistachio_mp/keymaps/via/keymap.c
Comment (drashna) [inline keyboards/pistachio_mp/keymaps/via/keymap.c:29]:
```suggestion
#define LAYOUT_via( \
         ECW,ECCW, K07, \
    K00, K01, K02, K03, \
    K04, K05, K06,      \
    K08, K09, K10, K11, \
    K12, K13, K14,      \
    K15,      K16, K17 ) { \
    { K00,   K01,   K02,   K03 }, \
    { K04,   K05,   K06,   K07 }, \
    { K08,   K09,   K10,   K11 }, \
    { K12,   K13,   K14,   ECCW }, \
    { K15,   ECW,   K16,   K17 }, \
}


const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
```

**Comment by 7-rate** [comment]
Thank you for your review.

**Review by drashna — APPROVED** [review]

**Comment by 7-rate** [comment]
If there are no problems, I would like you to merge them.

**Comment by drashna** [comment]
> If there are no problems, I would like you to merge them.

Generally, we want 2 approvals before a merge, because one set of eyes is likely to miss something.

**Review by tzarc — APPROVED** [review]

**Pull request merged**
